blocks/bricks/recurrent.py

```diff
@@ -179,11 +190,11 @@ def recurrent_apply(brick, application, application_call,
                             kwargs[state_name](state_name, batch_size,
                                                *args, **kwargs))
                 else:
-                    # TODO init_func returns 2D-tensor, fails for iterate=False
-                    kwargs[state_name] = (
-                        brick.initial_state(state_name, batch_size,
-                                            *args, **kwargs))
-                    assert kwargs[state_name]
+                    try:
+                        kwargs[state_name] = initial_states[state_name]
+                    except KeyError:
+                        raise KeyError(
+                            "no initial state for '{}'".format(state_name))
```

Comment:
May be we should add a link to a recurrent brick here. Like `No initial state for {} in brick {}`.